python/lsst/daf/butler/tests/_testRepo.py

```diff
@@ -341,17 +345,45 @@ def addDataIdValue(butler, dimension, value):
         The name of the dimension to gain a new value.
     value
         The value to register for the dimension.
+    **related
+        Any existing dimensions to be linked to ``value``.
+
+    Examples
+    --------
+
+    .. code-block::py
```

Comment:
My experience is that the function docstrings are a lot more visible than the how-to guide, but I don't want the guide to not have examples...

Reply:
Would it be possible to say "for example usage, see [URL here]" in the docstring? It just feels like every time we have docs in more than one place they get out of sync and we forget which one is right.

Reply:
Normal sphinx lets you read code snippets from a separate file so that it's guaranteed to be identical. I don't think I've every tried it from code sphinx docs but maybe it would work anyhow?

Reply:
I remember it working *very* badly in developer guide examples; evolution of the underlying code turned them into gibberish. I'll go with @mrawls's idea; it'll have the added benefit of making high-level documentation more visible.